finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

```diff
@@ -143,9 +148,9 @@ object Trace {
     val spanId = SpanId(rng.nextLong())
```

Comment:
I'm not an expert, and this isn't part of this change, but I recall a comment by @adriancole saying that a span-id of 0 is not valid, but it's possible that this `rng.nextLong()` could give us a 0. Is that only true in the high bits?

Reply:
`ThreadLocalRandom.current().nextLong(1,Long.MaxValue)` could provide non-zero positive values, but as I understood it all longs are valid _except_ 0.

Reply:
traceIdHigh is invalid as non-zero for sure (though using the aws-convertable scheme I mentioned, it is impossible to end up with a zero traceIdHigh unless you are tracing 1970 :) ).

Reply:
one thing to bear in mind is that negative is needed or else you won't get full random. if trying to avoid 0, you can loop while nextLong == 0. This is unlikely to loop twice

Reply:
For simplicity, I'd vote for looping while `nextLong == 0`. Another alternative, but one that is probably more work and not worth it, is 
```
var spanId = ThreadLocalRandom.current().nextLong(Long.MinValue, Long.MaxValue - 1)
if (0l <= spanId) {
  spanId += 1
}
```
FWIW, it's still not clear to me whether 0 is even illegal.